src/index.js

```diff
@@ -160,6 +160,7 @@ class Libp2p extends EventEmitter {
     log('libp2p is starting')
     try {
       await this._onStarting()
+      this._isStarted = true
```

Comment:
I think we should put this in `_onDidStart` as that keeps all post startup changes there.